sideboard/lib/_profiler.py

```diff
@@ -0,0 +1,271 @@
+"""
+Adds profiling tools and a web interface for viewing profiling results.
+
+The Sideboard profiler borrows heavily from the `CherryPy profiler
+<https://github.com/cherrypy/cherrypy/blob/master/cherrypy/lib/profiler.py>`_,
+but with a few added features and nicer formatting.
+
+ * Adds the ability to sort results by different columns.
+ * Adds the ability to cleanup profile data files.
+ * Uses a better naming scheme for profile data files.
+ * Uses `cProfile` instead of `profile` for better performance.
+
+Profiling data can be collected using the @profile decorator on functions and
+methods. The profiling results can be viewed at http://servername/profile/.
+
+Good candidates for profiling are the outermost functions that generate your
+web pages, usually exposed as cherrypy endpoints via @cherrypy.expose::
+
+    import cherrypy
+    from sideboard.lib import profile
+
+    class Root(object):
+        @cherrypy.expose
+        @profile
+        def index(self):
+            # Create and return the index page
+            return '<html/>'
+
+
+But any regular function can be profiled using the @profile decorator::
+
+    from sideboard.lib import profile
+
+    @profile
+    def some_interesting_function():
+       # Do some stuff
+
+
+The following config options control how the profiler operates, see
+configspec.ini for more details::
+
+    [cherrypy]
+    profiling.on = boolean(default=True)
+    profiling.path = string(default="%(root)s/data/profiler")
+    profiling.aggregate = boolean(default=False)
+    profiling.strip_dirs = boolean(default=False)
+
+"""
+import io
+import os
+import os.path
+import cProfile
+import pstats
+from datetime import datetime
+from functools import wraps
+from glob import glob
+
+import cherrypy
+from sideboard.config import config
+from sideboard.lib import entry_point, listify
+
+
+def _new_func_strip_path(func_name):
+    """
+    Adds the parent module to profiler output for `__init__.py` files.
+
+    Copied verbatim from cherrypy/lib/profiler.py.
+    """
+    filename, line, name = func_name
+    if filename.endswith('__init__.py'):
+        return os.path.basename(filename[:-12]) + filename[-12:], line, name
+    return os.path.basename(filename), line, name
+
+pstats.func_strip_path = _new_func_strip_path
+
+
+@entry_point
+def cleanup_profiler():
+    """
+    Deletes all `*.prof` files in the profiler's data directory.
+
+    This is useful when you've created tons of profile files that you're no
+    longer interested in. Exposed as a `sep` command::
+
+        $ sep cleanup_profiler
+
+    The profiler directory is specified in the config by::
+
+        [cherrypy]
+        profiling.path = 'path/to/profile/data'
+
+    """
+    profiling_path = config['cherrypy'].get('profiling.path', None)
```

Comment:
Minor style nitpick: ``None`` is already the default value returned by ``dict.get``, so ``d.get('foo')`` is already the same as ``d.get('foo', None)``.

Larger nitpick: it's actually impossible for ``config['cherrypy']['profiling.path']`` to ever not be defined, since our ``configspec.ini`` guarantees that it will have a value.  Therefore there's no need to use ``.get`` here or to have this if-statement.